addons/payment_stripe/static/src/js/payment_form.js

```diff
@@ -42,7 +43,250 @@ odoo.define('payment_stripe.payment_form', require => {
+                    country:'BE',
+                    currency: this.txContext.currencyName,
+                    total: {
+                        label: this.txContext.label,
+                        amount: this.txContext.minorAmount,
+                        pending: isShippingInformationRequired,
+                    },
+                    displayItems: isShippingInformationRequired ? [
+                        {
+                            label: 'Your order',
+                            amount: this.txContext.minorAmount,
+                        },
+                        {
+                            label: 'Delivery',
+                            amount: 0,
+                            pending: false,
+                        },
+                    ]:[],
+                    requestPayerName: true,
+                    requestPayerEmail: true,
+                    requestPayerPhone: true,
+
+                    requestShipping: isShippingInformationRequired,
+                });
+                self.StripeExpress = paymentRequest;
+                self.StripeExpress.acquirerId = paymentOptionId;
+
+                var elements = stripeJS.elements();
+                var prButton = elements.create('paymentRequestButton', {
+                    paymentRequest: paymentRequest,
+                    style: {
+                        paymentRequestButton: {
+                          type: 'buy',
+                          // One of 'default', 'book', 'buy', or 'donate'
+                          // Defaults to 'default'
+                        },
+                      },
+                });
+
+                // Check the availability of the Payment Request API first.
+                paymentRequest.canMakePayment().then(function(result) {
+                    if (result) {
+                        prButton.mount(`#o_stripe_express_checkout_container_${paymentOptionId}`);
+                    }
+                });
+
+                paymentRequest.on('paymentmethod', function(ev) {
+                    console.log(ev);
+                    var addresses = {}
+                    addresses['partner_id'] = parseInt(self.txContext.partnerId)
+                    addresses['billing_address'] = {
+                        name: ev.payerName,
+                        email: ev.payerEmail,
+                        phone: ev.payerPhone,
+                        street: ev.paymentMethod.billing_details.address.line1,
+                        street2: ev.paymentMethod.billing_details.address.line2 ? ev.paymentMethod.billing_details.address.line2 : false,
+                        zip: ev.paymentMethod.billing_details.address.postal_code,
+                        city: ev.paymentMethod.billing_details.address.city,
+                        country: ev.paymentMethod.billing_details.address.country,
+                        state: ev.paymentMethod.billing_details.address.state ? ev.paymentMethod.billing_details.address.state : false,
```

Comment:
Having a local `billing_address` variable would make this a lot more readable.

Reply:
Indeed, but those data are given by Stripe's javascript framework and we can't change the output.